dot/network/service.go

```diff
@@ -157,6 +157,15 @@ func NewService(cfg *Config) (*Service, error) {
 		bufPool = newSizedBufferPool(cfg.MinPeers*3, cfg.MaxPeers*3)
 	}
 
+	var blockResponseBufPool *sizedBufferPool
+	if cfg.noPreAllocate {
+		blockResponseBufPool = &sizedBufferPool{
+			c: make(chan *[maxMessageSize]byte, cfg.MinPeers*3),
+		}
+	} else {
+		blockResponseBufPool = newSizedBufferPool(cfg.MinPeers*3, cfg.MaxPeers*3)
+	}
```

Comment:
nit just some small simplifications 😉 

```suggestion
	poolSize := 3 * cfg.MaxPeers
	var preAlloc int
	if !cfg.noPreAllocate {
		preAlloc = 3 * cfg.MaxPeers
	}
	blockResponseBufPool := newSizedBufferPool(preAlloc, poolSize)
```